### PR TITLE
Fix helm name override

### LINKED ISSUE #22622: Helm values nameOverride and fullnameOverride are not p
### Official Helm Chart version

1.5.0 (latest released)

### Apache Airflow version

2.2.4 (latest released)

### Kubernetes Version

1.21

### Helm Chart configuration

In your values.yaml you can specify fullnameOverride or nameOverride. Neither is added to the names of the Deployment objects created.

# Provide a name to substitute for the full names of resources
fullnameOverride: "whatever"

# Provide a name to substitute for the name of the chart
nameOverride: "whatever"

Generated names will look like:

airflow-scheduler-c48f6d677-zc72b
airflow-webserver-7b6fc468fb-xq86l

In this example "airflow" is the release name. 

If fullnameOverride is set it should take precedence and the name should look like:

whatever-scheduler-c48f6d677-zc72b
whatever-webserver-7b6fc468fb-xq86l

If nameOverride is set and fullnameOverride is set then the name should be included with the release name. For example:

airflow-whatever-scheduler-c48f6d677-zc72b
airflow-whatever-webserver-7b6fc468fb-xq86l

### Docker Image customisations

_No response_

### What happened

When setting name overrides, the deployment objects do not include the overrides in deployment objects

### What you think should happen instead

Deployment objects should include the name overrides.

### How to reproduce

Set nameOverride or fullnameOverride and deploy to a cluster. Inspect your Deployments and find that the name is based on .Release.Name without including overrides.

### Anything else

The template function that should be used has already been added to _helpers.yaml: https://github.com/apache/airflow/blob/9e66d14dc1212e54c0eb22cc43a1eee764a8fed8/chart/templates/_helpers.yaml#L23

Someone just needs to go through the various *-deployment.yaml files and update the name to use the template function.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### LINKED ISSUE #22790: Helm deployment fails when postgresql.nameOverride is used
### Apache Airflow version

2.2.5 (latest released)

### What happened

Helm installation fails with the following config:

``` 
postgresql:
  enabled: true
  nameOverride: overridename
```

The problem is manifested in the `-airflow-metadata` secret where the connection string will be generated without respect to the `nameOverride`

With the example config the generated string should be:

`postgresql://postgres:postgres@myrelease-overridename:5432/postgres?sslmode=disable`

but the actual string generated is:
`postgresql://postgres:postgres@myrelease-overridename.namespace:5432/postgres?sslmode=disable`

### What you think should happen instead

Installation should succeed with correctly generated metadata connection string

### How to reproduce

To reproduce just set the following in values.yaml and attempt `helm install`

``` 
postgresql:
  enabled: true
  nameOverride: overridename
```

### Operating System

Ubuntu

### Versions of Apache Airflow Providers

_No response_

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

using helm with kind cluster

### Anything else

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by evanrosebrook**
This change completes the work required to allow nameOverride and fullnameOverride to function as expected. Kubernetes objects will now be created in a manner consistent with current helm conventions. 

Objects created with fullnameOverride specified will use the fullnameOverride to create object names.
If nameOverride is specified the value of nameOverride is combined with the chart name to create object names.
If nameOverride and fullnameOverride are both specified then fullnameOverride will take precedence.

This change fixes: https://github.com/apache/airflow/issues/22622

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
While I think it is good, it seems like a potentially breaking change? I am not sure if this is a problem though.

**Comment by evanrosebrook** [comment]
Within the deployment itself it is not breaking. However, if teams have built external monitoring that expects specific names that were changed, it could mess with that.

**Comment by potiuk** [comment]
> Within the deployment itself it is not breaking. However, if teams have built external monitoring that expects specific names that were changed, it could mess with that.

Yep. That exactly what I mean :). We **might**  not care about it as it is an internal detail. Or treat it as bugfix ("shoudl work like that always"). I am leaning towards the latter, but I will leave it to others to comment :)

File: chart/values.schema.json [hunk 1]
```diff
@@ -799,8 +799,8 @@
             "x-docsSection": "Airflow",
             "default": null,
             "examples": [
-                "- secretRef:\n    name: '{{ .Release.Name }}-airflow-connections'",
-                "- configMapRef:\n    name: '{{ .Release.Name }}-airflow-variables'"
+                "- secretRef:\n    name: '{{ include 'airflow.fullname' . }}-airflow-connections'",
```
Comment (jedcunningham) [inline chart/values.schema.json]: Even though it's an example, the quoting here seems like it'd be wrong. Maybe this?
```suggestion
                "- secretRef:\n    name: \"{{ include 'airflow.fullname' . }}-airflow-connections\"",
```

File: docs/helm-chart/production-guide.rst [hunk 2]
```diff
@@ -452,17 +452,17 @@ Here is the full list of secrets that can be disabled and replaced by ``_CMD`` a
 +-------------------------------------------------------+------------------------------------------+--------------------------------------------------+
 | Default secret name if secret name not specified      | Use a different Kubernetes Secret        | Airflow Environment Variable                     |
 +=======================================================+==========================================+==================================================+
-| ``<RELEASE_NAME>-airflow-metadata``                   | ``.Values.data.metadataSecretName``      | | ``AIRFLOW_CONN_AIRFLOW_DB``                    |
+| ``<RELEASE_NAME>-metadata``                   | ``.Values.data.metadataSecretName``      | | ``AIRFLOW_CONN_AIRFLOW_DB``                    |
```
Comment (jedcunningham) [inline docs/helm-chart/production-guide.rst]:
```suggestion
| ``<RELEASE_NAME>-metadata``                           | ``.Values.data.metadataSecretName``      | | ``AIRFLOW_CONN_AIRFLOW_DB``                    |
```

Think I got the right number of spaces :)

File: docs/helm-chart/production-guide.rst [hunk 3]
```diff
@@ -452,17 +452,17 @@ Here is the full list of secrets that can be disabled and replaced by ``_CMD`` a
...
 +-------------------------------------------------------+------------------------------------------+--------------------------------------------------+
 | ``<RELEASE_NAME>-fernet-key``                         | ``.Values.fernetKeySecretName``          | ``AIRFLOW__CORE__FERNET_KEY``                    |
 +-------------------------------------------------------+------------------------------------------+--------------------------------------------------+
 | ``<RELEASE_NAME>-webserver-secret-key``               | ``.Values.webserverSecretKeySecretName`` | ``AIRFLOW__WEBSERVER__SECRET_KEY``               |
 +-------------------------------------------------------+------------------------------------------+--------------------------------------------------+
-| ``<RELEASE_NAME>-airflow-result-backend``             | ``.Values.data.resultBackendSecretName`` | | ``AIRFLOW__CELERY__CELERY_RESULT_BACKEND``     |
+| ``<RELEASE_NAME>-result-backend``             | ``.Values.data.resultBackendSecretName`` | | ``AIRFLOW__CELERY__CELERY_RESULT_BACKEND``     |
```
Comment (jedcunningham) [inline docs/helm-chart/production-guide.rst]:
```suggestion
| ``<RELEASE_NAME>-result-backend``                     | ``.Values.data.resultBackendSecretName`` | | ``AIRFLOW__CELERY__CELERY_RESULT_BACKEND``     |
```

File: docs/helm-chart/production-guide.rst [hunk 4]
```diff
@@ -452,17 +452,17 @@ Here is the full list of secrets that can be disabled and replaced by ``_CMD`` a
...
 |                                                       |                                          | | ``AIRFLOW__CELERY__RESULT_BACKEND``            |
 +-------------------------------------------------------+------------------------------------------+--------------------------------------------------+
-| ``<RELEASE_NAME>-airflow-brokerUrl``                  | ``.Values.data.brokerUrlSecretName``     | ``AIRFLOW__CELERY__BROKER_URL``                  |
+| ``<RELEASE_NAME>-brokerUrl``                  | ``.Values.data.brokerUrlSecretName``     | ``AIRFLOW__CELERY__BROKER_URL``                  |
```
Comment (jedcunningham) [inline docs/helm-chart/production-guide.rst]:
```suggestion
| ``<RELEASE_NAME>-brokerUrl``                          | ``.Values.data.brokerUrlSecretName``     | ``AIRFLOW__CELERY__BROKER_URL``                  |
```

**Review by jedcunningham — COMMENTED** [review]
I also lean toward this not being a breaking change. That said, we might consider keeping, for example, `airflow-metadata` instead of stripping the `airflow-` portion. That would then allow the names to stay consistent with `fullnameOverride`, if someone needed the "old" names. We should also add a section noting this to UPDATING.md.

**Comment by evanrosebrook** [comment]
`airflow-` has been restored to secrets: `airflow-metadata`, `-airflow-results-backend` and configmap: `-airflow-config` and a comment about compatibility has been added to UPDATING.rst

File: chart/UPDATING.rst [hunk 5]
```diff
@@ -43,6 +43,9 @@ Default Airflow image is updated to ``2.2.5``
 
 The default Airflow image that is used with the Chart is now ``2.2.5``, previously it was ``2.2.4``.
 
+Resources naming has been updated to support ``nameOverride`` and ``fullnameOverride``. The default name is now derived from .Release.Name and chart name.
```
Comment (jedcunningham) [inline chart/UPDATING.rst:49]:
```suggestion
Support for ``nameOverride`` and ``fullnameOverride``
"""""""""""""""""""""""""""""""""""""""""""""""""""""

Resources naming has been updated to support ``nameOverride`` and ``fullnameOverride``. The default name is now derived from .Release.Name and chart name.
```

File: docs/helm-chart/production-guide.rst [hunk 6]
```diff
@@ -452,17 +452,17 @@ Here is the full list of secrets that can be disabled and replaced by ``_CMD`` a
...
 +-------------------------------------------------------+------------------------------------------+--------------------------------------------------+
 | ``<RELEASE_NAME>-fernet-key``                         | ``.Values.fernetKeySecretName``          | ``AIRFLOW__CORE__FERNET_KEY``                    |
 +-------------------------------------------------------+------------------------------------------+--------------------------------------------------+
 | ``<RELEASE_NAME>-webserver-secret-key``               | ``.Values.webserverSecretKeySecretName`` | ``AIRFLOW__WEBSERVER__SECRET_KEY``               |
 +-------------------------------------------------------+------------------------------------------+--------------------------------------------------+
-| ``<RELEASE_NAME>-airflow-result-backend``             | ``.Values.data.resultBackendSecretName`` | | ``AIRFLOW__CELERY__CELERY_RESULT_BACKEND``     |
+| ``<RELEASE_NAME>-airflow-result-backend``                     | ``.Values.data.resultBackendSecretName`` | | ``AIRFLOW__CELERY__CELERY_RESULT_BACKEND``     |
```
Comment (jedcunningham) [inline docs/helm-chart/production-guide.rst]:
```suggestion
| ``<RELEASE_NAME>-airflow-result-backend``             | ``.Values.data.resultBackendSecretName`` | | ``AIRFLOW__CELERY__CELERY_RESULT_BACKEND``     |
```

Oops, now we have extra spaces :)

File: docs/helm-chart/production-guide.rst [hunk 7]
```diff
@@ -452,17 +452,17 @@ Here is the full list of secrets that can be disabled and replaced by ``_CMD`` a
 +-------------------------------------------------------+------------------------------------------+--------------------------------------------------+
 | Default secret name if secret name not specified      | Use a different Kubernetes Secret        | Airflow Environment Variable                     |
 +=======================================================+==========================================+==================================================+
-| ``<RELEASE_NAME>-airflow-metadata``                   | ``.Values.data.metadataSecretName``      | | ``AIRFLOW_CONN_AIRFLOW_DB``                    |
+| ``<RELEASE_NAME>-airflow-metadata``                           | ``.Values.data.metadataSecretName``      | | ``AIRFLOW_CONN_AIRFLOW_DB``                    |
```
Comment (jedcunningham) [inline docs/helm-chart/production-guide.rst]:
```suggestion
| ``<RELEASE_NAME>-airflow-metadata``                   | ``.Values.data.metadataSecretName``      | | ``AIRFLOW_CONN_AIRFLOW_DB``                    |
```

**Review by jedcunningham — COMMENTED** [review]
Last few things I see 🍺

**Comment by jedcunningham** [comment]
Pre-install hooks strike again 😔. Playing with this a bit, this doesn't work for upgrades... We have some objects that are created with pre-install hooks, and those names don't get updated. Might not be comprehensive, but these are the secrets I see problems with:
- broker-url
- fernet_key
- redis-password

Try updating a release on main to your branch and you should be able to replicate it.

I can't think of an easy solution to this unfortunately.

**Comment by evanrosebrook** [comment]
Would it not be appropriate to add the `pre-upgrade` hook to those objects that have the `pre-install` hook?

**Comment by jedcunningham** [comment]
Unfortunately no, as they aren't mean to be changed, and by default they get a randomly generated value. Helms `lookup` also isn't great - it doesn't work with dry runs/diffs.

**Comment by evanrosebrook** [comment]
Seems like a `lookup` based solution is the only viable option. I understand that it won't work for `dry-run`s etc, but I haven't been able to think of anything else. Any other ideas?

I figure the lookup will be used to copy the secret from the old resource to the new resources. If I am not mistaken?

**Comment by gerardo-garcia** [comment]
Any possibility that this pull request can be reopened and merged? The issue behind it is still valid.

**Comment by potiuk** [comment]
> Any possibility that this pull request can be reopened and merged? The issue behind it is still valid.

Sure. You an open your PR - use the code here and lead it to completion. No problem if you would like to take on the task. The PR was closed because there was no author who would follow that and lead it to completion - but anyone - like you - can do it.

**Pull request closed without merging**